src/algo/thermal-loop/l500-thermal-loop.cpp

```diff
@@ -0,0 +1,92 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include "l500-thermal-loop.h"
+#include "../../l500/l500-private.h"
+
+namespace librealsense {
+namespace algo {
+namespace thermal_loop {
+namespace l500 {
+
+const int thermal_calibration_table::id = 0x317;
+
+thermal_calibration_table::thermal_calibration_table( const std::vector< byte > & data,
+                                                      int resolution )
+    : _resolution( resolution )
+{
+    float const * header_ptr = (float *)( data.data() + sizeof( ivcam2::table_header ) );
+
+    auto expected_size = sizeof( ivcam2::table_header ) + sizeof( thermal_table_header )
+                       + sizeof( temp_data ) * resolution;
+
+    if( data.size() != expected_size )
+        throw std::runtime_error( librealsense::to_string()
+                                  << "data size (" << data.size()
+                                  << ") does not meet expected size " << expected_size );
+
+    _header = *(thermal_table_header *)( header_ptr );
+
+    auto data_ptr = (temp_data *)( data.data() + sizeof( ivcam2::table_header )
+                                   + sizeof( thermal_table_header ) );
+    vals.assign( data_ptr, data_ptr + resolution );
+}
+
+double thermal_calibration_table::get_current_thermal_scale( const double & hum_temp ) const
+{
+    auto scale = vals[_resolution - 1].scale;
+
+    // curr temp is under minimum
+    if( hum_temp <= _header.min_temp )
+    {
+        scale = vals[0].scale;
+    }
+    else
+    {
+        auto temp_range = _header.max_temp - _header.min_temp;
+        // there are 29 bins between min and max temps so its divides to 30 equals intervals
+        auto interval = temp_range / ( thermal_calibration_table::_resolution + 1 );
+
+        for( double temp = _header.min_temp, index = 0; index < _resolution;
```

Comment:
Shouldn't index start at 1?

Reply:
And obviously, go to `<=`?

Reply:
Actually, no -- I see now that index 0 is at the END of the first interval... confusing, maybe need to add a comment:

```
//      |--|--|--|...|--|
//     min 0  1  2...29 max
```

Reply:
Done